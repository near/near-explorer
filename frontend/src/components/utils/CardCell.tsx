import { Row, Col, Card } from "react-bootstrap";

export interface Props {
  className?: string;
  title: React.ReactElement | string;
  imgLink?: string;
  text: React.ReactElement | string;
}

export default ({ title, imgLink, text, className }: Props) => (
  <Card className={`card-cell ${className || ""}`}>
    <Card.Body>
      <Row noGutters>
        <Col xs="auto" md="12" className="card-cell-title align-self-center">
          {imgLink && <img src={imgLink} className="card-cell-title-img" />}
          {title}
        </Col>
        <Col
          xs="auto"
          md="12"
          className="ml-auto card-cell-text align-self-center"
        >
          {text}
        </Col>
      </Row>
    </Card.Body>
    <style jsx global>{`
      .card-cell {
        border-style: solid;
        border-color: #e6e6e6;
        border-width: 0 0 0 2px;
        border-radius: 0;
      }

      .card-cell-title {
        text-transform: uppercase;
        letter-spacing: 1.8px;
        color: #999999;
        font-family: BentonSans;
        font-size: 14px;
        font-weight: 500;
      }

      .card-cell-title-img {
        width: 12px !important;
        margin-right: 8px;
        margin-top: -3px;
      }

      .card-cell-text {
        font-family: BwSeidoRound;
        font-size: 18px;
        font-weight: 500;
        color: #24272a;
      }

      .card-cell a {
        font-weight: 500;
        color: #6ad1e3;
        text-decoration: none !important;
      }

      .card-cell a:hover {
        color: #6ad1e3;
      }
    `}</style>
  </Card>
);
