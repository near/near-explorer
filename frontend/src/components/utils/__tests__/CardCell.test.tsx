import renderer from "react-test-renderer";

import CardCell from "../CardCell";

describe("<CardCell />", () => {
  it("renders", () => {
    expect(
      renderer.create(<CardCell title="title" text="text" />)
    ).toMatchSnapshot();
  });
});
