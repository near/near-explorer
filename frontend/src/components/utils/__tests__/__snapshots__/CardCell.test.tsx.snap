// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCell /> renders 1`] = `
<div
  className="card-cell  card"
>
  <div
    className="card-body"
  >
    <div
      className="row no-gutters"
    >
      <div
        className="card-cell-title align-self-center col-md-12 col-auto"
      >
        title
      </div>
      <div
        className="ml-auto card-cell-text align-self-center col-md-12 col-auto"
      >
        text
      </div>
    </div>
  </div>
</div>
`;
